Additions/changes to Get Started (#642)
spatialos/UnrealGDK: SpatialGDK/Documentation/content/get-started/build-unreal-fork.md

> # Get started: 2 - Get and build the SpatialOS Unreal Engine Fork

Author: Not 100% sure about the wording of this. Are we calling it the SpatialOS Unreal Engine fork, or the GDK for Unreal fork? etc.

> # Get started: 2 - Get and build the SpatialOS Unreal Engine fork

Author: Not 100% sure about the wording of this. Are we calling it the SpatialOS Unreal Engine fork, or the GDK for Unreal fork? etc.

Reviewer: either is fine but I think SpatialOS looks cooler

Author: SpatialOS _does_ look cooler.

Reviewer 2: The official term in the Canonology is: "Unreal Engine fork" .
I think we do need to use "GDK" or "SpaitialOS" when we first introduce it but we need to make sure it's consistent and not change it without reason.
"The SpatialOS Unreal Engine fork" is easier to use outside of the context of the GDK for Unreal docs because otherwise you get "the GDK for Unreal Unreal Engine fork" and that's nasty.

Reviewer 2: Adding "SpatialOS Unreal Engine fork" to the canonology. Let me know if we change it back.

The GDK for Unreal extends Unreal Engine's networking capabilities at its core. To use it you first need to build our SpatialOS fork of Unreal Engine from source.

> 

Author: We could just reword it, the first bit probably is superfluous. Would like @girayimprobable's input on this one though as I don't want to remove potentially important info. 
```suggestion
To use the SpatialOS GDK for Unreal, you first need to build our SpatialOS fork of Unreal Engine.
```

> 

Reviewer 2: If you use the first edit, above, I think you need a comma after "To use it".

> 

Reviewer 3: Funnily enough `core` is a technical term for UE4, however it's not where the network layer resides. We were definitely using it more for its descriptive properties. Safe to remove.

> 

Author: Changed to
```
To use the SpatialOS GDK for Unreal, you first need to build the SpatialOS fork of Unreal Engine.
```

### Step 1: Unreal Engine EULA

> 

Reviewer 2: So, we originally didn't have "accept the invite to join" but someone got trolled by not realising they would get an email invite to join, so we added this in. I can't remember who it was.

Author: Good shout. I thought the Unreal documentation mentioned the org invite, but it doesn't seem to.

To get access to our fork, you need to link your GitHub account to a verified Epic Games account, and agree to the Unreal Engine End User License Agreement ([EULA](https://www.unrealengine.com/en-US/eula)). You cannot use the GDK without doing this first. To do this, see the [Unreal Engine documentation](https://www.unrealengine.com/en-US/ue4-on-github).

### Step 2: Getting the Unreal Engine fork source code and Unreal Linux cross-platform support

> > You should clone the Unreal Engine fork into your root directory to avoid file path length errors. <br/>For example: C:\GitHub\UnrealEngine

Reviewer 2: ```suggestion
> **TIP:** Clone the Unreal Engine fork into your root directory to avoid file path length errors. <br/>For example: C:\GitHub\UnrealEngine
```
Should the clone tip come after the clone step??!

1\.  Clone the [Unreal Engine fork](https://github.com/improbableio/UnrealEngine/tree/4.20-SpatialOSUnrealGDK) repository by opening a terminal and running either of these commands: 

> You may get a 404 from this link. See  the instructions above, under _Unreal Engine EULA_, on how to get access to this repository. 

> |     |     |

Author: Not as far as I'm aware. Markdown tables are weird and I hate them. We could use HTML but I don't know if Improbadoc plays nicely with it.

> |     |     |

Reviewer: HTML is definitely not worth the fuss, don't worry about it.

| --- | --- |
| HTTPS | `git clone https://github.com/improbableio/UnrealEngine.git -b 4.20-SpatialOSUnrealGDK` |
| SSH |`git clone git@github.com:improbableio/UnrealEngine.git -b 4.20-SpatialOSUnrealGDK`

2\.  To build Unreal server-workers for SpatialOS deployments you need to build the Unreal Engine fork targeting Linux. This requires cross-compilation of your SpatialOS project and the Unreal Engine fork.

In Unreal's [Compiling for Linux](https://wiki.unrealengine.com/Compiling_For_Linux) documentation, in the **getting the toolchain** section, click **v11 clang 5.0.0-based** to download the **v11_clang-5.0.0-centos7.zip** archive, then unzip this file into a suitable directory.

### Step 3: Adding environment variables

> You need to add two [environment variables](https://docs.microsoft.com/en-us/windows/desktop/shell/user-environment-variables): one to set the path to the Unreal Engine fork directory, and another one to set the path to the Linux cross-platform support directory.

Reviewer: ```suggestion
You need to add two [environment variables](https://docs.microsoft.com/en-us/windows/desktop/procthread/environment-variables): one to set the path to the Unreal Engine fork directory, and another one to set the path to the Linux cross-platform support directory.
```

Reviewer: Changed the documentation this links to because, while I think the doc you chose is clearer and more concise, it only talks about user environment variables, and we're asking users to set system environment variables. This might cause confusion.

Author: Good catch. Think I just slammed the first link to environment variable documentation in there.

1. Go to **Control Panel > System and Security > System > Advanced system settings > Advanced > Environment variables**.
2. Create a system variable named **UNREAL_HOME**.
3. Set the variable value to the path to the directory you cloned the Unreal Engine fork into.
4. Restart your terminal and run `echo %UNREAL_HOME%` (Command Prompt) or `echo$Env:UNREAL_HOME` (PowerShell). If you have registered the environment variable correctly, this returns the path to the directory you cloned the Unreal Engine fork into. If it doesn’t, check that you’ve set the environment variable correctly.
5. Create a system variable named **LINUX_MULTIARCH_ROOT**.
6. Set the variable value to the path to the directory of your unzipped Linux cross compilation toolchain.
7. Restart your terminal and run `echo %LINUX_MULTIARCH_ROOT%` (Command Prompt) or `echo $Env:LINUX_MULTIARCH_ROOT` (PowerShell). If you have registered the environment variable correctly, this returns the path you unzipped `v11_clang-5.0.0-centos7.zip` into. If it doesn’t, check that you’ve set the environment variable correctly.

> ### Step 4: Building Unreal Engine

Reviewer 2: Just realised we are gerund-tastic (building, adding - lots of ings) here - wondering why we did this, when we are trying to be all imperative (build, add).

Author: Did an "ing" purge. See the latest commit.

> 1. Open **Explorer** and navigate to the directory you cloned the Unreal Engine fork into.

Reviewer 2: Is this MMoS?  Gosh. Embarrassing we were using the old term?!
